packages/dashboard/src/components/tasks/task-inspector.tsx

```diff
@@ -0,0 +1,243 @@
+          setTaskLogs(null);
+        }
+        setTaskState(task);
+      })();
+    });
+    return () => sub.unsubscribe();
+  }, [rmf, waypoint]);
+
+  React.useEffect(() => {
+    if (!rmf) {
+      return;
+    }
+    const sub = AppEvents.taskSelect
+      .pipe(
+        switchMap((selectedTask) =>
+          selectedTask ? rmf.getTaskStateObs(selectedTask.booking.id) : of(null),
+        ),
+      )
+      .subscribe(setTaskState);
+    return () => sub.unsubscribe();
+  }, [rmf]);
+
+  React.useEffect(() => {
+    AppEvents.taskSelect.next(task);
+  }, [task]);
+
+  const profile = React.useContext(UserProfileContext);
+
+  const taskCancellable =
+    taskState &&
+    profile &&
+    Enforcer.canCancelTask(profile) &&
+    taskState.status &&
+    !['canceled', 'killed', 'completed', 'failed'].includes(taskState.status);
+
+  const handleCancelTaskClick = React.useCallback<React.MouseEventHandler>(async () => {
+    if (!taskState) {
+      return;
+    }
+    try {
+      if (!rmf) {
+        throw new Error('tasks api not available');
+      }
+      await rmf.tasksApi?.postCancelTaskTasksCancelTaskPost({
+        type: 'cancel_task_request',
+        task_id: taskState.booking.id,
+      });
+      appController.showAlert('success', 'Successfully cancelled task');
+      AppEvents.taskSelect.next(null);
+    } catch (e) {
+      appController.showAlert('error', `Failed to cancel task: ${(e as Error).message}`);
+    }
+  }, [appController, taskState, rmf]);
+
+  return (
+    <>
+      <Grid container wrap="nowrap" direction="column" height="100%">
+        <Dialog
+          open={open}
+          onClose={onClose}
+          aria-labelledby="alert-dialog-title"
+          aria-describedby="alert-dialog-description"
+          fullWidth
+          maxWidth="md"
+        >
+          <Grid container spacing={2}>
+            <Grid item xs={12}>
+              <DialogTitle id="scroll-dialog-title" align="center">
+                {task?.booking.id}
+              </DialogTitle>
+            </Grid>
+            <Grid container>
+              <Grid item xs={2} ml={7}>
+                <Typography>Status:</Typography>
+              </Grid>
+              <Grid item xs={3}>
+                <Typography align="right">{task?.status}</Typography>
+              </Grid>
+            </Grid>
+            <Grid container>
+              <Grid item xs={2} ml={7}>
+                <Typography>Start Waypoint:</Typography>
+              </Grid>
+              <Grid item xs={3}>
+                <Typography align="right">{waypoint.start}</Typography>
+              </Grid>
+            </Grid>
+            <Grid container>
+              <Grid item xs={2} ml={7}>
+                <Typography>Finish Waypoint:</Typography>
+              </Grid>
+              <Grid item xs={3}>
+                <Typography align="right">{waypoint.finish}</Typography>
+              </Grid>
+            </Grid>
+          </Grid>
```

Comment:
As mentioned in a comment below, we should migrate these fields down to `TaskInfo`, based on our initial designs on figma